src/sagemaker/jumpstart/notebook_utils.py

```diff
@@ -283,166 +310,146 @@ def _generate_jumpstart_model_versions(  # pylint: disable=redefined-builtin
-                    f"is failing: {(model_manifest.model_id, model_manifest.version)}."
+        model_specs = JumpStartModelSpecs(
+            json.loads(
+                sagemaker_session.read_s3_file(
+                    get_jumpstart_content_bucket(region), model_manifest.spec_key
                 )
-            copied_filter_2 = copy.deepcopy(filter)
-
-            model_specs = accessors.JumpStartModelsAccessor.get_model_specs(
-                region=region,
-                model_id=model_manifest.model_id,
-                version=model_manifest.version,
             )
+        )
 
-            model_specs_keys = set(model_specs.__slots__)
+        for val in possible_spec_keys:
+            if hasattr(model_specs, val):
+                manifest_specs_cached_values[val] = getattr(model_specs, val)
 
-            unrecognized_keys -= model_specs_keys
-            unrecognized_keys_for_single_spec = possible_spec_keys - model_specs_keys
-            unrecognized_keys.update(unrecognized_keys_for_single_spec)
+        _populate_model_filters_to_resolved_values(
+            manifest_specs_cached_values,
+            model_filters_to_resolved_values,
+            model_filters,
+        )
+        _put_resolved_booleans_into_filter(copied_filter_2, model_filters_to_resolved_values)
 
-            for val in possible_spec_keys:
-                if hasattr(model_specs, val):
-                    manifest_specs_cached_values[val] = getattr(model_specs, val)
+        copied_filter_2.eval()
 
-            _populate_model_filters_to_resolved_values(
-                manifest_specs_cached_values,
-                model_filters_to_resolved_values,
-                model_filters,
-            )
-            _put_resolved_booleans_into_filter(copied_filter_2, model_filters_to_resolved_values)
+        if copied_filter_2.resolved_value != BooleanValues.UNEVALUATED:
+            if copied_filter_2.resolved_value == BooleanValues.TRUE or (
+                BooleanValues.UNKNOWN and list_incomplete_models
+            ):
+                return (model_manifest.model_id, model_manifest.version)
+            return None
 
-            copied_filter_2.eval()
+        raise RuntimeError(
+            "Filter expression in unevaluated state after using values from model specs. "
+            "Model ID and version that is failing: "
+            f"{(model_manifest.model_id, model_manifest.version)}."
+        )
 
-            if copied_filter_2.resolved_value != BooleanValues.UNEVALUATED:
-                if copied_filter_2.resolved_value == BooleanValues.TRUE or (
-                    BooleanValues.UNKNOWN and list_incomplete_models
-                ):
-                    yield (model_manifest.model_id, model_manifest.version)
-                continue
+    max_memory_bytes = int(100 * 1e6)
+    average_memory_bytes_per_thread = int(25 * 1e3)
+    max_workers = int(max_memory_bytes / average_memory_bytes_per_thread)
```

Comment:
move to a constant at the top of the module please:

```py
MAX_SEARCH_WORKERS = int(100 * 1e6 / 25 * 1e3)  # max 100 MB total memory, 25kB per thread)
```

Also, isn't that _a lot_ of threads?

Reply:
yes it is a lot of threads. since this is an i/o intensive task, compared to the amount of compute being done, this shouldn't degrade the performance (and doesn't based on my testing).

Reply:
I see, please confirm with SDK team though.